### DISCUSSION TITLE
Don't stop maintaining this lol

### DISCUSSION THREAD

**Discussion by rly0nheart** [hamdanal/rich-argparse#102] [Ideas]
Dude you need to keep on maintaining this😅, I just stumbled upon it while looking for a way to colourise my argparse help. It hasn't disappointed me so far. If by any chance you figure that you can't actively maintain it, I'd be more than happy to help with it. I've started updating all my projects to use **rich-argpase** .

Thanks for this man!♥️

**Comment by hamdanal** [comment]
Thank you for your kind words, I am glad you like this project. Don't worry, I don't have any plans to abandon it :)

**Comment by bulletmark** [comment]
@hamdanal , I have a few projects which use sub-parsers and they look great if I plug this in but  I am worried about the durability of this project wrt to future versions of Python/argparse. Are you really confident you can maintain compatibility across all current and future Python versions given how intimately this code integrates?

I just looked through `argparse` source changes and I noticed that your recently [added a significant commit](https://github.com/python/cpython/commit/de1428f8c234a8731ced99cbfe5cd6c5c719e31d) there so I guess it is clear you know argparse internals pretty well but it just feels to me that module hacks a little too much of the internal implementation of argparse. Naively at least as a first attempt, I would try implementing this just with simple regexps in an overridden `_print_message()`. Did you try that or other approaches first? I am keen to hear your experience/comments.

**Comment by hamdanal** [comment]
This project is always tested with pre-release versions of python. That and argparse being extremely stable make me not worry much about compatibility issues.

> it just feels to me that module hacks a little too much of the internal implementation of argparse

All add-ons to argparse use part of its private api, simply because it has no usable public api. Look at the famous argcomplete project for example.

> I would try implementing this just with simple regexps in an overridden _print_message()

That won’t work for every use case, people do weird stuff in their code and expect it to work (the description text of the program is a free form text, you can easily fool any regular expression). It might work for your use case though, you’d have to try it out and you’ll need to test it every time you update your cli.

**Comment by bulletmark** [comment]
@hamdanal are you aware that Python 3.14, which is released in a few days, adds colored output to `argparse` usage/help messages?

**Reply by hamdanal** [reply]
Yes
